src/test/java/org/junitpioneer/jupiter/ReportEntryExtensionTests.java

```diff
@@ -78,8 +79,9 @@ void repeatedAnnotation_logEachKeyValuePairAsIndividualEntry() {
 
 		assertThat(results)
 				.hasNumberOfReportEntries(3)
-				.withValues("suddenly there came a tapping", "As if some one gently rapping",
-					"rapping at my chamber door");
+				.andThen(entry -> assertThat(entry.getValue())
+						.isIn("suddenly there came a tapping", "As if some one gently rapping",
+							"rapping at my chamber door"));
```

Comment:
I don't like this new assertions. In my opinion it's much more complicated and less readable. The user is now forced to write an additional lamba expression and another assertion.

I think it would be more readable with something like `.hasNumberOfReportEntries(3).andTheyAre("suddenly..", "As...", "rapping...")`ore something like this.

Reply:
This is an open-ended method that lets you chain your own assertions at the end of the 'common' ones.

For example: Stopwatch extension publishes entries with "Test took X ms to run". You can't test for that, because you can only test for exact values (right now). But with the `andThen` method, you can write:
```
assertThat(results)
    .hasSingleReportEntry()
    .andThen(entry ->
        assertThat(entry.getValue()).matches("Test took [0-9]+ ms to run")
    );
```

Reply:
We can keep adding new assertion methods for new test-cases when a feature needs it (like `StopwatchExtension`) or have a generic method "exposing" the things we want to assert (exceptions, report entries).